department-of-veterans-affairs/caseflow
# Add CAVC Power of Attorney admin action

Bumps [CASEFLOW-107]

### Description
Allows cavc users to add poa admin action.

### Acceptance Criteria
- [x] The following task actions are added to SendCavcRemandProcessedLetterTask
  - [x]  "Assign POA Clarification task to CAVC team" - Creates a CavcPoaClarificationTask assigned to the CavcLitigationSupportTeam as a child of the SendCavcRemandProcessedLetterTask
- [x] Provide Instructions and context for this action appears on the submit admin actions page
- [x] Users have standard task actions on the poa clarification task (assign, reassign, complete, cancel)

### Testing Plan
1. Log in as the CAVC team admin and assign any open `SendCavcRemandProcessedLetterTask` to yourself
1. Click the task action dropdown and confirm new poa clarification action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a poa clarification task assigned to the cavc team is created as a child of the send letter task
1. Confirm you no longer have actions available to reassign or complete the send letter task
```ruby
Appeal 934 (evidence_submission)                       ID   STATUS      ASGN_BY                ASGN_TO                UPDATED_AT
└── RootTask                                           2841 on_hold                            Bva                    2020-11-23 16:07:47 UTC
    └── DistributionTask                               2842 on_hold                            Bva                    2020-11-23 16:07:47 UTC
        └── CavcTask                                   2843 on_hold                            Bva                    2020-11-23 16:07:47 UTC
            └── SendCavcRemandProcessedLetterTask      2844 on_hold                            CavcLitigationSupport  2020-11-23 17:01:12 UTC
                └── SendCavcRemandProcessedLetterTask  3500 on_hold     CAVC_LIT_SUPPORT_ADMIN CAVC_LIT_SUPPORT_ADMIN 2020-11-23 17:14:51 UTC
                    └── CavcPoaClarificationTask       3501 assigned    CAVC_LIT_SUPPORT_ADMIN CavcLitigationSupport  2020-11-23 17:15:17 UTC
```

### User Facing Changes

 Workflow Step|UI
 ---|---
New task action|<img width="897" alt="Screen Shot 2020-11-23 at 12 01 19 PM" src="https://user-images.githubusercontent.com/45575454/100001088-97305600-2d90-11eb-9aae-7547d423666a.png">
Modal|<img width="515" alt="Screen Shot 2020-11-23 at 12 13 48 PM" src="https://user-images.githubusercontent.com/45575454/100001096-9a2b4680-2d90-11eb-9f68-e4cca9a5af85.png">
Modal validation|<img width="511" alt="Screen Shot 2020-11-23 at 12 14 02 PM" src="https://user-images.githubusercontent.com/45575454/100001103-9e576400-2d90-11eb-90cb-c9609d703b98.png">
Success & updated task snapshot|<img width="999" alt="Screen Shot 2020-11-23 at 12 14 55 PM" src="https://user-images.githubusercontent.com/45575454/100001120-a2838180-2d90-11eb-85b2-54545821d43a.png">
POA clarification admin actions|<img width="651" alt="Screen Shot 2020-11-23 at 1 37 52 PM" src="https://user-images.githubusercontent.com/45575454/100001474-381f1100-2d91-11eb-8e9c-9f3f61a90645.png">
POA clarification assignee actions|<img width="652" alt="Screen Shot 2020-11-23 at 1 38 06 PM" src="https://user-images.githubusercontent.com/45575454/100001482-3b1a0180-2d91-11eb-9427-b6b82d51fb81.png">
